### PR TITLE
Ss dt api setup

### PR DESCRIPTION
**Pull request by snsebro**
## Description

-Made call to Wikipeadia api through provided proxy endpoint
-Rendered location pins on map based on response from api call
-Added inconify package to project for easy acess icon components 

## Acceptance Criteria

- [x] Pins should be 24px x 24px in size

- [x] Pins should have a little down caret on the bottom edge of the pin that points to the location of the place

- [x] Pins should show a scaled-down (object-fit: cover style) thumbnail image in cases when a thumbnail image is present

- [x] Pins should render a default pin image when no thumbnail is returned

## Type of Changes

|     | Type                       |
| --- | -------------------------- |
|     | :bug: Bug fix              |
| ✓   | :sparkles: New feature     |
|     | :hammer: Refactoring       |
|     | :100: Add tests            |
|  ✓   | :link: Update dependencies |
|     | :scroll: Docs              |

### Before
image

### After
image

## Testing Steps / QA Criteria

Updating the location coordinates in the api call and adjusting the centering parameters in the GoogleMap React component should render pins for neraby locations returned from the api. Check the components section in dev tools to see that the api response has been saved to state.

### CODE REVIEW

File: src/pages/Map.js [hunk 1]
```diff
@@ -1,25 +1,49 @@
...
-  lat: 21.694,
-  lng: 71.7979,
+  lat: 45,
+  lng: -123.456,
 };
 
 const zoomLevel = 8;
 
-export default function Map() {
+export default function Map({ locations }) {
```
Comment (rachel-fischoff) [inline src/pages/Map.js:16]: I was able to see the locations in the state in app & they are passed down as props! The icons look great! The only comment is a get a lot of warnings (12) in my console. I've tried to google it & it's a google maps API bug. It seems to do with how props are being passed down? Did y'all get these warnings too?

**Review by rachel-fischoff — APPROVED** [review]
Hi y'all! Everything looks great. I'm really enjoying the pins and how they are rendered. I did encounter a bug! My console is filled with about 12 warnings. I've done a little research and it looks like it's bc of the Google Maps API & how pass down props. Did y'all get those as well? https://github.com/google-map-react/google-map-react/issues/740


Screen Shot 2021-01-16 at 8 10 19 PM

**Comment by segdeha** [comment]
I left the same comment on the other pair's PR, but would you mind renaming the PR to something more descriptive than "Ss dt api setup". Think of what would be helpful for the person reviewing the PR to get them oriented to the change quickly. Thanks!

Also, you can delete the help text in the template when you make your PRs:

image

**Comment by segdeha** [comment]
I'll look at the code next, but just looking at the UI, I'm a little confused by this icon:

image

Is that mean to be a default icon? It looks like a mistake because it appears there's an image behind the black pin icon?

Curious whether that was an intentional design decision or if something's not rendering correctly?

File: src/App.js [hunk 2]
```diff
@@ -8,11 +8,29 @@ import Nav from './components/Nav';
 import './App.css';
 
 function App() {
+  const [locations, setLocations] = useState([]);
+
+  var url = 'https://segdeha.com/api/nearby.php?lat=45&lng=-123.456';
```
Comment (segdeha) [inline src/App.js]: These days, you shouldn't ever need to use `var`. In this case, you probably want a `const`.

File: src/App.js [hunk 3]
```diff
@@ -8,11 +8,29 @@ import Nav from './components/Nav';
...
+  useEffect(() => {
+    fetch(url)
+      .then(function (response) {
+        return response.json();
+      })
+      .then(function (response) {
+        let pages = response.query.pages;
+        setLocations(pages);
+      })
+      .catch(function (error) {
```
Comment (segdeha) [inline src/App.js]: This whole `catch` block can be written as the following:

    .catch(console.log);

Passing in a reference to `console.log` will cause the `error` parameter to be passed to the function. Functional programming FTW!

File: src/App.js [hunk 4]
```diff
@@ -8,11 +8,29 @@ import Nav from './components/Nav';
 import './App.css';
 
 function App() {
+  const [locations, setLocations] = useState([]);
+
+  var url = 'https://segdeha.com/api/nearby.php?lat=45&lng=-123.456';
+
+  useEffect(() => {
+    fetch(url)
+      .then(function (response) {
```
Comment (segdeha) [inline src/App.js]: This works fine, but a more modern looking way to write this would be with ES6-style functions such as the following:

    .then(response => {
      return response.json();
    })

Comment (segdeha) [reply src/App.js]: There's no functional difference in this case, but it just looks a little funny now to see the `function` keyword.

File: src/App.js [hunk 5]
```diff
@@ -8,11 +8,29 @@ import Nav from './components/Nav';
 import './App.css';
 
 function App() {
+  const [locations, setLocations] = useState([]);
```
Comment (segdeha) [inline src/App.js:11]: Way to remember to set a default value. I always forget! 😅

File: src/api.js [hunk 6]
```diff
@@ -0,0 +1,28 @@
+var url = 'https://en.wikipedia.org/w/api.php';
```
Comment (segdeha) [inline src/api.js]: Let's modernize this file by changing the `var` statements to either `const` or `let` statements, depending on whether you need to assign to them more than once. Thanks!

File: src/api.js [hunk 7]
```diff
@@ -0,0 +1,28 @@
...
+var params = {
+  action: 'query',
+  generator: 'geosearch',
+  prop: 'coordinates|pageimages',
+  ggscoord: '37.7891838|-122.4033522',
+  format: 'json',
+};
+
+url = url + '?origin=*';
+Object.keys(params).forEach(function (key) {
```
Comment (segdeha) [inline src/api.js]: It doesn't change the functionality, but I'm inclined to be a stickler for ES6-style functions mainly because not using them could raise questions from someone reviewing a take home test in an interview.

So, this would change to the following:

    Object.keys(params).forEach(key => {
        url += '&' + key + '=' + params[key];
    });

File: src/api.js [hunk 8]
```diff
@@ -0,0 +1,28 @@
...
+  url += '&' + key + '=' + params[key];
```
Comment (segdeha) [inline src/api.js]: Oh, another thing is you can make this code simpler/easier to read by using ES6 templates as so:

    url += `&${ key }=${ params[key] }`;

File: src/api.js [hunk 9]
```diff
@@ -0,0 +1,28 @@
...
+});
+
+fetch(url)
+  .then(function (response) {
```
Comment (segdeha) [inline src/api.js]: Prefer ES6 functions, so:

    .then(response => {
        return response.json();
    })

File: src/api.js [hunk 10]
```diff
@@ -0,0 +1,28 @@
...
+    return response.json();
+  })
+  .then(function (response) {
+    var pages = response.query.pages;
+    for (var page in pages) {
```
Comment (segdeha) [inline src/api.js]: Here's a case where `var` is actually an error. Well, it doesn't _throw_ an error, but that's JavaScript being nice to you. `var` doesn't allow for declaration in lexical scopes. That's pretty jargon-y, but what's happening here under the hood is that in the interpretation phase, a variable `page` is being created at the top of the function scope, then being assigned to in the execution phase.

That's probably _not_ what you thought was happening, which I expect is more like what `let` does: creating a variable `page` that is only available inside the `for` loop.

TL;dr, change this `var` to `let` 😅

File: src/api.js [hunk 11]
```diff
@@ -0,0 +1,28 @@
...
+Object.keys(params).forEach(function (key) {
+  url += '&' + key + '=' + params[key];
+});
+
+fetch(url)
+  .then(function (response) {
+    return response.json();
+  })
+  .then(function (response) {
+    var pages = response.query.pages;
```
Comment (segdeha) [inline src/api.js]: Prefer `const` to `var` here

File: src/api.js [hunk 12]
```diff
@@ -0,0 +1,28 @@
...
+    for (var page in pages) {
+      console.log(response.json());
+    }
```
Comment (segdeha) [inline src/api.js]: Wait, is this file even being used? 😅 

I expected to see a `return` statement here, then I went looking for where you import `api.js` but didn't see it anywhere. Is this file in use?

Comment (segdeha) [reply src/api.js]: If the file isn't being imported anywhere, please delete it. As you can see, I spent quite a bit of time reviewing a file that I'm pretty sure isn't even in use. That's...not great.

File: src/components/LocationPin.js [hunk 13]
```diff
@@ -0,0 +1,13 @@
+import React from 'react';
+import './LocationPin.css';
+
+export default function LocationPin({ img }) {
+  return (
+    <div className="pin">
+      <div
+        className="pin-fill"
+        style={{ backgroundImage: `url(${img})` }}
```
Comment (segdeha) [inline src/components/LocationPin.js:9]: I love to see the ES6 templates! 👏

File: src/pages/Map.js (hunk 1 above)
Comment (segdeha) [reply src/pages/Map.js:16]: @rachel-fischoff were you running it locally? I don't see console warnings when running the app from the preview deploy 🤔

File: src/pages/Map.js [hunk 14]
```diff
@@ -1,25 +1,49 @@
...
     <div className="map">
       <GoogleMapReact
-        bootstrapURLKeys={{
-          key: API_KEY,
-        }}
+        bootstrapURLKeys={{ key: API_KEY }}
         defaultCenter={location}
         defaultZoom={zoomLevel}
-      ></GoogleMapReact>
+        yesIWantToUseGoogleMapApiInternals
```
Comment (segdeha) [inline src/pages/Map.js:23]: This prop name always cracks me up 😂

**Review by segdeha — CHANGES_REQUESTED** [review]
Unless I'm mistaken, it appears that `api.js` isn't used anywhere, is that right? If so, please delete it.

I also left a few comments about JavaScript style that could be more modern (ES6-style). We don't _strictly_ have to fix those, but this kind of thing would raise eyebrows in an interview session, so I suggest getting more comfortable with ES6-style syntax as your default.

From a product standpoint, the code is behaving as-expected, so I'm happy to approve once these changes are made. Nice work!

File: src/api.js (hunk 12 above)
Comment (snsebro) [reply src/api.js]: Apologies and noted! I was using this as a working file to get some logic down and forgot to remove

File: src/App.js (hunk 4 above)
Comment (snsebro) [reply src/App.js]: Agreed, I took this block from the docs and didn't remeber to update to ES6 syntax. Thanks for catching these

**Review by segdeha — APPROVED** [review]
These changes look good, thanks!

I think this is ready to merge. We can talk as a team about what to do about pins that at the same lat/lng
